airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,205 @@
+  const { control, handleSubmit, reset, watch } = useForm({
+    defaultValues: {
+      configJson: JSON.stringify(dagParams.configJson), // Ensure it's a string in the form control
+      logicalDate: dagParams.logicalDate,
+      runId: dagParams.runId,
+    },
+  });
+
+  useEffect(() => {
+    reset({
+      configJson: JSON.stringify(dagParams.configJson),
+      logicalDate: dagParams.logicalDate,
+      runId: dagParams.runId,
+    });
+  }, [dagParams, reset]);
+
+  const onSubmit = () => {
+    onTrigger();
+  };
+
+  const hasFormChanged = () => {
+    const currentValues = {
+      configJson: watch("configJson"),
+      logicalDate: watch("logicalDate"),
+      runId: watch("runId"),
+    };
+
+    return (
+      currentValues.configJson !== JSON.stringify(dagParams.configJson) ||
+      currentValues.logicalDate !== dagParams.logicalDate ||
+      currentValues.runId !== dagParams.runId
+    );
+  };
+
+  const { colorMode } = useColorMode();
+
+  return (
+    <>
+      <Dialog.CloseTrigger />
+
+      <VStack align="stretch" gap={2}>
+        <Button
+          mb={9}
+          onClick={() => setShowDetails(!showDetails)}
+          variant="outline"
+          width="full"
+        >
+          {showDetails ? "Hide Advanced Options" : "Show Advanced Options"}
+        </Button>
+
+        {showDetails ? (
+          <VStack align="stretch" gap={3}>
+            <Box>
+              <Text fontSize="md" mb={2}>
+                Logical date
+              </Text>
+              <Controller
+                control={control}
+                name="logicalDate"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="yyyy-mm-ddThh:mm"
+                    size="sm"
+                    type="datetime-local"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box>
+              <Text fontSize="md" mb={2}>
+                Run ID
+              </Text>
+              <Controller
+                control={control}
+                name="runId"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="Run id, optional - will be generated if not provided"
+                    size="sm"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box mb={9}>
+              <Text fontSize="md" mb={2}>
+                Configuration JSON
+              </Text>
+              <Controller
+                control={control}
+                name="configJson"
+                render={({ field }) => (
+                  <Box>
+                    <CodeMirror
+                      {...field}
+                      basicSetup
+                      extensions={[json(), autocompletion(), lineNumbers()]}
+                      height="200px"
+                      onChange={(value) => {
```

Comment:
We might need to debounce this function

Reply:
I am trying this, facing a weird problem, The value i am typing in the code box post debouncing the function, is not getting converted.

Reply:
We can leave it for now and add later.